docs/source/modules/models.rst

```diff
@@ -23,6 +23,19 @@ To avoid race conditions when using :ref:`hydra_launcher_plugins` that may run m
    The weights for all pretrained models that are provided by `autrainer` can be automatically downloaded using the
    :ref:`autrainer fetch <cli_autrainer_fetch>` CLI command or the :meth:`~autrainer.cli.fetch` CLI wrapper function.
 
+To optionally use model, optimizer, or scheduler checkpoints, the following attributes can be set in any model configuration file:
+
+* :attr:`model_checkpoint`: The path to the model checkpoint file.
+  If the output dimensions between the checkpoint and the model configuration do not match,
+  the checkpoint will be loaded without the last linear layer.
+* :attr:`optimizer_checkpoint`: The path to the optimizer checkpoint file.
+  If the output dimensions between the checkpoint and the model configuration do not match,
+  the checkpoint will be loaded without the last linear layer, similar to the model checkpoint.
+  If no model checkpoint is provided, :attr:`skip_last_layer` is automatically set to ``True``.
```

Comment:
I'm wondering if default should be false, in case it's more of a continued training, maybe this is an equally likely use case as the change of dataset?

Reply:
Continuing training is already implemented, if a run exists with the same configuration and only differs by the number of iterations (smaller), it is automatically continued (as long as `continue_training` is set to true in the main configuration).

`continue_training`, however, begins training from the last state. Passing checkpoints allows for more granularity from which state to start training (+ allowing the dynamic adjustment of hyperparameters which is not possible while continuing training).

But i am totally open to changing `skip_last_layer` to `True` by default, such that this has to be explicitly disabled, what do you prefer?

Reply:
No strong opinion here. From a user perspective, the dangers of misuse are in the default `True` case that they believe they are continuing the training as they are not aware of the flag, which they are not and it would be obvious to know. In the default `False` case it might be that an error is being thrown because of the mismatch. I am aware of the `continue-training` option, which indeed kind of covers the same idea, but for instance is not usable if a run is cancelled prior to finishing say due to time limitations.

Reply:
The main problem is when a user only passes an `optimizer_checkpoint` and there would be a potential shape mismatch, no error is thrown at all during loading the state dict. However in the first optimization step, there could be a potential shape mismatch for example when using Adam. This would then throw an error which seems completely unrelated with loading the optimizer checkpoint and is therefore completely masked.
Thats what my intention was to set `skip_last_layer` always to `True` in this case, preventing this issue from happening in the first place.

I do see the that there are dangers of misuse, but to discover checkpointing, a user has to read the docs anyways, so i feel like if we properly explain the `skip_last_layer` option, it should be fine either way.


For me, both `True` and `False` could be seen as intuitive, when considering that our "intended use case" of checkpointing is to first train a model on dataset A and later fine-tune it on dataset B, then `skip_last_layer` should be `True` by default, but still allowing the user to explicitly keep the same classification head if desired (and possible).
Right now, if the shape does not match, we also skip the layer, also if `skip_last_layer` is `False`. Should we keep this then if it is `True` by default or just fail?

What would you prefer?

Reply:
I tend towards explicitly failing when something's not working rather than do something that the user might not want us to do

Reply:
I do prefer this as well. Right now, we have `skip_last_layer: True` as default value, which is think is justified to distinguish checkpointing and continuing training. If the user explicitly sets `skip_last_layer: False` or no last layer can be identified, and this results in an actual shape mismatch, we just fail and not try to do some hidden control flow to recover.